### DISCUSSION TITLE
Raspberry pi pico nrf24l01 driver error

### DISCUSSION THREAD

**Discussion by UmutDemirel06** [micropython/micropython#11601] [Libraries & Drivers] [answered]
Hi, I am trying to use the NRF24L01 module just to see how it works. I am using MicroPython to code, obviously, but I am encountering an error. Here is the error I am getting:

![1](https://github.com/micropython/micropython/assets/59196975/f68e9b45-ef39-477c-80fd-ee20f270c4ee)

This is the code that I tried for testing:

https://github.com/tinkertechtrove/pico-pi-playing/blob/main/nrf24l01-spi/button.py

And this is where I got the library:

https://github.com/micropython/micropython-lib/tree/master/micropython/drivers/radio/nrf24l01

I am new to this hobby, and I just want the NRF module to work properly. Any kind of suggestion is welcome.

**Comment by jimmo** [comment]
First guess would be something wrong with the wiring or the module. Can you describe (or post a photo) of how it's connected up?

**Reply by UmutDemirel06** [reply]
https://youtu.be/aP8rSN-1eT0 ı connected as this vid shows

**Reply by jimmo** [reply]
Sure, but that doesn't help figure out what's wrong with your setup. Can you post a photo of yours?

Do you have a multimeter, can you check voltages etc on the nRF module?

**Reply by UmutDemirel06** [reply]
sure ı will do that as soon as ı can

**Comment by peterhinch** [comment]
I suggest you start with [the official test script](https://github.com/micropython/micropython-lib/blob/master/micropython/drivers/radio/nrf24l01/nrf24l01test.py) which has been tested on Picos. The code comments list the way to wire them up.

**Reply by UmutDemirel06** [reply]
hi ı tried to do the test but 
![image](https://github.com/micropython/micropython/assets/59196975/1c6a8f06-4c86-4509-bff7-f6c7b606940a)
it's at this stage and ı don't know how to continue

**Reply by UmutDemirel06** [reply]
this is the wireing that ı did 
![20230904_192746](https://github.com/micropython/micropython/assets/59196975/7b0fdd95-73cb-4280-99f3-51cb28f55f77)

**Reply by steveh127** [reply]
Your wiring is wrong. For pin numbers you are using the board pin numbers but what you actually need are the GPIO Pin numbers. It's rare for board numbers and GPIO numbers to coincide. Consult the PICO pinout diagram for the correct pin allocations. They don't need to move far.

Also you have your 3.3v connection to the 3.3v EN pin not the 3.3v out pin on at least one of your boards.

Worth double checking module connections, it's impossible to see if they are correct from the photo.

I've tested the NRF24L901 board using the micropython-lib drivers and test program as suggested by Peter and it works as advertised with correctly connected pins.

**Reply by UmutDemirel06** [reply]
ı am currently fixing the wiring thanks for your reply but ı dont know how to do the responder initiator thing can you help me with that

**Reply by UmutDemirel06** [reply]
ı finally ran the test but it's success=0 failure=16

**Comment by peterhinch** [comment]
Install `nrf24l01test.py` on both boards. Run
```python
import nrf24l01test
```
on each board. It will print out the wiring that you should be using and will tell you how to run the test.

**Reply by UmutDemirel06** [reply]
it did tell me and ı fixed and checked the wiring but ı didnt under stand how to run initiator and responder thing

**Comment by peterhinch** [comment, marked as answer]
You run
```python
>>> nrf24l01test.responder()
```
on one board, then
```python
>>> nrf24l01test.initiator()
```
on the other.
